src/account/operations/address_generation.rs

```diff
@@ -83,58 +83,91 @@ impl AccountHandle {
+        #[cfg(feature = "stronghold")]
+        if let SecretManager::Stronghold(stronghold_secret_manager) = &mut *self.secret_manager.write().await {
+            stronghold_secret_manager.read_stronghold_snapshot().await?;
+        }
+
+        let addresses = match *self.secret_manager.read().await {
+            #[cfg(feature = "ledger_nano")]
+            SecretManager::LedgerNano(_) | SecretManager::LedgerNanoSimulator(_) => {
+                // If we don't sync, then we want to display the prompt on the ledger with the address. But the user
+                // needs to have it visible on the computer first, so we need to generate it without the
+                // prompt first
+                if !options.metadata.syncing {
                     let mut changed_metadata = options.metadata.clone();
+                    // Change metadata so ledger will not show the prompt the first time
                     changed_metadata.syncing = true;
-                    let addresses = self
-                        .secret_manager
+                    let mut addresses = Vec::new();
+
+                    for address_index in address_range {
+                        #[cfg(feature = "events")]
+                        {
+                            // Generate without prompt to be able to display it
+                            let address = self
+                                .secret_manager
+                                .read()
+                                .await
+                                .generate_addresses(
+                                    account.coin_type,
+                                    account.index,
+                                    address_index..address_index + 1,
+                                    options.internal,
+                                    changed_metadata.clone(),
+                                )
+                                .await?;
+                            self.event_emitter.lock().await.emit(
+                                account.index,
+                                WalletEvent::LedgerAddressGeneration(AddressData {
+                                    address: address[0].to_bech32(bech32_hrp.clone()),
+                                }),
+                            );
+                        }
+                        // Generate with prompt so the user can verify
+                        let address = self
+                            .secret_manager
+                            .read()
+                            .await
+                            .generate_addresses(
+                                account.coin_type,
+                                account.index,
+                                address_index..address_index + 1,
+                                options.internal,
+                                options.metadata.clone(),
+                            )
+                            .await?;
```

Comment:
What's different between the two generations ?

Reply:
The first one will just generate the address and return it direclty, the second one shows the address on the screen of the ledger nano and requires physical approval to continue

Reply:
Sorry if I'm being stupid but the two generation are exactly the same so why can't we just do the emit under the feature ?

Reply:
The first one has `changed_metadata.clone(),`  where syncing is set to true, which makes the ledger to not display the address and return without user approval
Then the event with the address gets emitted and Firefly can display the address, so when the second generation is called , with `options.metadata.clone()` and the ledger displays the address, the user can compare it with the address displayed in Firefly

Reply:
Ah damn I don't know how I missed changed_metadata != options.metadata, sorry about that 🤦🏻